XenAdmin/Controls/XenSearch/GroupingControl.cs

```diff
@@ -403,6 +403,7 @@ void button_MouseUp(object sender, MouseEventArgs e)
         {
             draggedButton = null;
             dragging = false;
+            Setup();

```

Comment:
Yes, the priority here is to avoid somehow Setup being called multiple times (my other comment was rhetoric :) )